### PR TITLE
Introduce creepLog, config based logging of creeps

### PR DESCRIPTION
**Pull request by TooAngel**
`config.creepLog.roles` and `config.creepLog.rooms` to filter specific
roles and rooms.

### CODE REVIEW

File: src/prototype_creep.js [hunk 1]
```diff
@@ -42,6 +42,7 @@ Creep.prototype.moveToMy = function(target, range) {
     visualizer.showSearch(search);
   }
 
+  this.creepLog('moveToMy search:' + JSON.stringify(search));
```
Comment (mschultheiss83) [inline src/prototype_creep.js]: this.creepLog('moveToMy search:', JSON.stringify(search));
// use join on string (+) or array of strings? (,)

File: src/logging.js [hunk 2]
```diff
@@ -5,3 +5,13 @@ Room.prototype.log = function(...messages) {
...
 };
+
+Creep.prototype.creepLog = function(...messages) {
+  if (config.debug.creepLog.roles.indexOf(this.memory.role) < 0) {
+    return;
+  }
+  if (config.debug.creepLog.rooms.indexOf(this.room.name) < 0) {
+    return;
+  }
+  console.log(`${Game.time} ${this.room.name.rpad(' ', 6)} ${this.name.rpad(' ', 20)} ${this.pos} ${messages.join(' ')}`);
```
[the remark this answers is not in the document]
Comment (TooAngel) [reply src/logging.js]: Good point, thx

File: src/prototype_creep.js (hunk 1 above)
Comment (TooAngel) [reply src/prototype_creep.js]: :+1 thx

**Comment by TooAngel** [comment]
Surely we can improve logging even further, any third party solution in mind?

I think we can debug single creeps much better with this system, I'm tired of adding and removing specific log lines for specific rooms and roles and the `creepLog` helps a lot in that regard.

**Comment by mschultheiss83** [comment]
same here logging i love just by changing in my local config debuging same roles in different rooms...

**Comment by mschultheiss83** [comment]
and everywhere where a creepLog makes it into the master we could later change to the debug or verbose mode....

third party solution: only one... elasticsearch

**Pull request merged**